builtin/logical/database/path_config_connection.go

```diff
@@ -377,6 +378,15 @@ func (b *databaseBackend) connectionReadHandler() framework.OperationFunc {
 		delete(config.ConnectionDetails, "private_key")
 		delete(config.ConnectionDetails, "service_account_json")
 
+		// If the auth_type is not gcp_iam we remove the field from the read response
+
+		if authType, ok := config.ConnectionDetails["auth_type"]; ok {
+			if authType != connutil.AuthTypeGCPIAM {
+				delete(config.ConnectionDetails, "use_private_ip")
+				delete(config.ConnectionDetails, "use_psc")
+			}
+		}
```

Comment:
My apologies! This isn't actually needed after all. These won't be returned in the response because connection details will only contain what was set in the write request.

Reply:
I just reverted my commit.